schemas/core/resource.go

```diff
@@ -1,9 +1,7 @@
 package core
 
-type Resource interface {
-	GetResourceType() *ResourceType
-	GetSchema() *Schema
-	GetSchemaExtensions() map[string]*Schema
-
+// ResourceTyper is the interface implemented by types that can hold Common resource attribute and are can return it ResourceType
+type ResourceTyper interface {
+	ResourceType() *ResourceType
 	GetCommon() *Common
```

Comment:
Why not simply `Common()` ?

Do you mean if I refactor ?

Reply:
We cannot use `Common()` because its name will collide with `Common` struct member.
For the same reason we have `GetSchema()` instead of `Schema()`

Reply:
You're right ... Hmmm :\

I propose to rename `Common` struct field into `CommonAttributes` or something similar.
And then rename `GetCommon()` to simply `Common()`.

Reply:
I agree, so I'm going to fix right now.

Reply:
done https://github.com/fabbricadigitale/scimd/pull/28/commits/bf4e50869dde6ff1081716139e67a861891f313e !